docs/user_guide/SIMP_Administration/Classification_and_Data/SIMP_Compliance_Engine.inc

```diff
@@ -0,0 +1,111 @@
+.. _SIMP Compliance Engine:
+
+The SIMP Compliance Engine
+--------------------------
+
+The :term:`SIMP Compliance Engine` allows users to enable both Puppet class
+parameter enforcement as well as code-level validation of both class and
+defined type parameters in compliance with regulatory and internal policies.
+
+This capability is provided by the `simp-compliance_markup`_ Puppet module and
+will be renamed at some point in the future to `simp-compliance_engine` for
+clarity.
+
+.. IMPORTANT::
+
+   Your system must be configured to use at least :term:`Hiera` version 5 for
+   the Compliance Engine to function properly.
+
+   SIMP 6.3+ supports this capability out of the box but earlier versions may
+   need to `perform a Hiera migration`_.
+
+Profile Enforcement
+^^^^^^^^^^^^^^^^^^^
+
+Enabling the SIMP Compliance Engine
+"""""""""""""""""""""""""""""""""""
+
+In the Puppet implementation, the Compliance Engine is a :term:`Hiera backend`
+that provides data from :term:`SIMP compliance profile` data sets for use in
+configuring Puppet classes via :term:`automatic parameter lookup`.
+
+The following example demonstrates how to enable the SIMP Compliance Engine:
+
+.. code-block:: yaml
+
+   ---
+   version: 5
+   defaults:
+     datadir: "data"
+     data_hash: "yaml_data"
+   hierarchy:
+     - name: "SIMP Compliance Engine"
+       lookup_key: "compliance_markup::enforcement"
+
+Remember that hierarchy lookups happen from top to bottom and stop at the first
+match. This means that if the SIMP Compliance Engine is in enforcing mode and
+is at the top of the hierarchy, then it cannot be overridden except by an
+:term:`ENC`. See the `Hiera config file syntax`_ page for additional
+information.
+
+Enforcing a Specific Profile
+""""""""""""""""""""""""""""
+
+The selection of profiles to enforce is based on the ``lookup_key`` entry in
+the ``hiera.yaml`` configuration as shown in the previous section.
+
+In this case, the ``lookup_key``, is ``compliance_markup::enforcement`` and
+should be set in Hiera as an Array of compliance profiles that you would like
+to apply, in priority order.
+
+At this time, SIMP CE comes with two publicly available profiles, ``disa_stig``
+and ``nist_800_53_rev4``.
+
+If you wanted to conform with the requirements of the :term:`DISA STIG`, you
+would apply the ``disa_stig`` profile as shown below:
+
+.. code-block:: yaml
+
+   ---
+   compliance_markup::enforcement:
+     - "disa_stig"
+
+The next run of Puppet on the nodes to which this Hiera data applies would then
+have their parameters set to meet the STIG requirements as mapped by parameter
+in the associated compliance profile.
+
+Additional profiles may be made available by various Puppet modules, such as
+those provided by :term:`SIMP EE`.
+
+Code-Level Compliance Validation
```

Comment:
This is where I am confused. 
1) Is Validation turned on by default?   This implies a default directory but not really that it is turned on by default.
2) If it is turned on by default, it should go before enforcement.  Because this comes after Enforcement, I might think I need to put the lookup key in hiera if I wasn't famliar with it.  
3) compliance_markup::validation is empty be default.  I am getting reports without setting this.  What is it comparing against?

Reply:
Validation is enabled if you set up Hiera correctly and yes, you do need to put the corresponding validation key in Hiera.

For `3.`, are you testing on a fresh system or something that was running an older version? It's possible that validation was enabled previously if you already have a configured system.

Reply:
I put validation after enforcement because the majority of questions that we get are about how to enforce settings.

A lot of people don't even know about the validation capabilities and I wanted them to be able to discover it instead of being confused by it when looking for how to enforce. It was kind of difficult to achieve.

Reply:
So I wasn't sure if I had compliance n previously or not, so I removed all the reports, made sure I had nothing in hiera, and ran puppet again and the reports came back.

Reply:
I will defer to your greater knowledge of the customer base.  I know, where I was coming from, I would not want enforce, unless I was designing a new system.  I would use validate to see where I differed and use "notes" to generate a mitigation section for that variable so I remember why I changed it.

Reply:
@jeannegreulich I tried making the two sections more clearly separate, hopefully that will help!